public/app/percona/integrated-alerting/components/Table/Filter/Filter.utils.ts

```diff
@@ -0,0 +1,143 @@
+import { UrlQueryMap, UrlQueryValue } from '@grafana/data';
+import { getValuesFromQueryParams } from 'app/percona/shared/helpers/getValuesFromQueryParams';
+import { ExtendedColumn, FilterFieldTypes } from '..';
+import { ALL_LABEL, ALL_VALUE, SEARCH_INPUT_FIELD_NAME, SEARCH_SELECT_FIELD_NAME } from './Filter.constants';
+
+export const getQueryParams = (columns: ExtendedColumn[], queryParams: UrlQueryMap) => {
+  const customTransform = (params: UrlQueryValue): string | undefined => {
+    if (params !== undefined && params !== null) {
+      return params.toString();
+    }
+    return undefined;
+  };
+  const queryKeys = columns.map((column) => ({ key: column.accessor as string, transform: customTransform }));
+  queryKeys.push({ key: SEARCH_INPUT_FIELD_NAME, transform: customTransform });
+  queryKeys.push({ key: SEARCH_SELECT_FIELD_NAME, transform: customTransform });
+  const params = getValuesFromQueryParams(queryParams, queryKeys);
+  return params ?? {};
+};
+
+export const buildObjForQueryParams = (columns: ExtendedColumn[], values: Record<string, any>) => {
+  let obj: Record<string, any> = {
+    [SEARCH_INPUT_FIELD_NAME]: values[SEARCH_INPUT_FIELD_NAME],
+    [SEARCH_SELECT_FIELD_NAME]: values[SEARCH_SELECT_FIELD_NAME]?.value ?? values[SEARCH_SELECT_FIELD_NAME],
+  };
+  if (obj[SEARCH_INPUT_FIELD_NAME] && !obj[SEARCH_SELECT_FIELD_NAME]) {
+    obj[SEARCH_SELECT_FIELD_NAME] = ALL_VALUE;
+  }
+  if (!obj[SEARCH_INPUT_FIELD_NAME] && obj[SEARCH_SELECT_FIELD_NAME]) {
+    obj[SEARCH_SELECT_FIELD_NAME] = undefined;
+  }
+  columns.forEach((column) => {
+    const accessor = column.accessor as string;
+
+    if (column.type === FilterFieldTypes.RADIO_BUTTON) {
+      if (values[accessor]) {
+        if (values[accessor] === ALL_VALUE) {
+          obj[accessor] = undefined;
+        } else {
+          obj[accessor] = values[accessor].toString();
+        }
+      }
+    }
+    if (column.type === FilterFieldTypes.DROPDOWN) {
+      if (values[accessor]) {
+        if (values[accessor]?.value === ALL_VALUE || values[accessor] === ALL_VALUE) {
+          obj[accessor] = undefined;
+        } else {
+          obj[accessor] = values[accessor]?.value ? values[accessor]?.value.toString() : values[accessor].toString();
+        }
+      }
+    }
+  });
+  return obj;
```

Comment:
A bit like above, these ifs can be made simpler

Reply:
simplified